core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3015,6 +3015,10 @@ public void setAttribute(AlluxioURI path, SetAttributeContext context)
     boolean rootRequired = options.hasOwner();
     // for chgrp, chmod
     boolean ownerRequired = (options.hasGroup()) || (options.hasMode());
+    // for other attributes
+    boolean writeRequired = options.hasPersisted() || options.hasPinned()
```

Comment:
when does this list of operations need to be updated? What is not included in this list of operations?

Reply:
This is is everything beside permission related updates. It needs to be updated when a new field is added to the option / inode.

Reply:
will this be hard to maintain? Should it be checked the other way around, meaning, if it is a permission change, then only owner is required, otherwise, write is required?

It will be very easy to forget to update this list of non-permission related updates.

Reply:
It's is hard to maintain either way. This original issue happens because we added set owner and set mode and we did not exclude them from checking write permission. I agree that if we forget to update it should at least check write permission, but we can still have bugs if we add new attributes which requires different permissions to set.